js/src/services/vue.service.js

```diff
@@ -0,0 +1,103 @@
+import Vue from 'vue';
+import atkInlineEdit from '../components/inline-edit.component';
+import itemSearch from '../components/item-search.component';
+
+
+let atkComponents = {
+  'atk-inline-edit' : atkInlineEdit,
+  'atk-item-search' : itemSearch,
+};
+
+/**
+ * Singleton class
+ * Create Vue component.
+ */
+class VueService {
+
+  static getInstance() {
+    return this.instance;
+  }
+
+  constructor() {
+    if(!VueService.instance){
+      this.vues = [];
+      this.eventBus = new Vue();
+      VueService.instance = this;
+    }
+    return VueService.instance;
+  }
+
+  /**
+   * Created a Vue component and add it to the vues array.
+   *
+   * @param name
+   * @param component
+   * @param data
+   */
+  createAtkVue(name, component, data) {
+    this.vues.push({name: name, instance: new Vue({
+          el: name,
+          components: {[component]: atkComponents[component]},
+          data: {item:data},
+          methods: {
+            getData: function() {
+              return this.item
+            }
+          },
+          // provide method to our child component.
+          // child would need to inject a method to have access.
+          provide: function() {
+            return {
+              getRootData: this.getData
+            };
+          }
+        }
+      )}
+    );
+  }
+
+  /**
+   * Create a Vue instance from an external src component definition.
+   *
+   * @param name
+   * @param component
+   * @param data
+   */
+  createVue(name, component, data) {
+    this.vues.push({name: name, instance: new Vue({
+          el: name,
+          components: {[component]: component},
+          data: {item:data},
+          methods: {
+            getData: function() {
+              return this.item
+            }
+          },
+          // provide method to our child component.
+          // child would need to inject a method to have access.
+          provide: function() {
+            return {
+              getRootData: this.getData
+            };
+          }
+        }
+      )}
+    );
+  }
+
+  /**
+   * Emit an event to the eventBus.
+   * Listener to eventBus can respond to emitted event.
+   *
+   * @param event
+   * @param data
+   */
+  emitEvent(event, data = {}) {
```

Comment:
unclear what this does and how to use it. No doc or example.

Reply:
This would allow for component define in seperate Vue instance to be able to talk to each other. 
For example, I use this is saasty for updating the add-ons category when using the item-search component of the add-on market page. The button emit an event that tell the item-search component the category needed to include in it's query.
Been said, the mechanism is in place but I do not have proper use case yet to use it.

Reply:
yes, but i mean - this need documentation.